pkg/prometheus/server/operator.go

```diff
@@ -1138,7 +1179,31 @@ func (c *Operator) configResStatusCleanup(ctx context.Context, p *monitoringv1.P
 	}); err != nil {
 		return fmt.Errorf("listing all ServiceMonitors from cache failed: %w", err)
 	}
+	if getErr != nil {
+		return getErr
+	}
 
+	// Remove bindings from all podMonitors which reference the workload.
+	if err := c.promInfs.ListAll(labels.Everything(), func(obj any) {
```

Comment:
small typo, big effect :)
```suggestion
	if err := c.pmonInfs.ListAll(labels.Everything(), func(obj any) {
```

Reply:
done the changes